### PR TITLE
Replace template field amendment_paragraph with json field

### PR DESCRIPTION
**Pull request by reiterl**
Update model, actions and tests.

### CODE REVIEW

**Comment by r-peschke** [comment]
Please write tests with filled amendment_paragraph, each with valid html and invalid html
- meeting.import
- meeting.clone
- motion.create
- motion.update

File: openslides_backend/action/actions/motion/mixins.py [hunk 1]
```diff
@@ -31,3 +33,15 @@ def is_submitter(self, submitter_ids: List[int], state_id: int) -> bool:
...
         )
+
+
+class AmendmentParagraphHelper:
+    def validate_amendment_paragraph(self, instance: Dict[str, Any]) -> None:
+        if not isinstance(instance["amendment_paragraph"], dict):
+            raise ActionException("Wrong amendment_paragraph except dict")
+        for key, html in instance["amendment_paragraph"].items():
+            if not str.isdigit(key):
+                raise ActionException(f"amendment_paragraph {key} not allowed.")
```
Comment (jsangmeister) [inline openslides_backend/action/actions/motion/mixins.py:44]: I would rather put this directly in the schema - you cannot use the `optional_properties` then, but have to enter the schema manually into the `additional_optional_properties`, but it's cleaner and more uniform than checking the schema here in the code.

**Review by jsangmeister — CHANGES_REQUESTED** [review]

File: openslides_backend/action/actions/motion/mixins.py (hunk 1 above)
Comment (reiterl) [reply openslides_backend/action/actions/motion/mixins.py]: I moved both check into the schema.

**Comment by reiterl** [comment]
I have added the missing import and checker code and have added some tests for them.

File: openslides_backend/action/actions/motion/create.py [hunk 2]
```diff
@@ -42,6 +42,10 @@ class MotionCreate(AmendmentParagraphHelper, MotionCreateBase):
         additional_optional_fields={
             "workflow_id": optional_id_schema,
             "submitter_ids": id_list_schema,
+            "amendment_paragraph": {
+                "type": "object",
+                "propertyNames": {"pattern": POSITIVE_NUMBER_REGEX},
```
Comment (jsangmeister) [inline openslides_backend/action/actions/motion/create.py:47]: Also restrict the values here (currently, a value of e.g. `null` or `1` is also valid, when only strings shoul be valid). I would suggest to use `patternProperties` in combination with `additionalProperties: false`.

I would also extract this partial jsonschema definition and reuse it in `motion.update`.

File: openslides_backend/models/checker.py [hunk 3]
```diff
@@ -562,6 +564,21 @@ def get_enum_from_collection_field(
...
+    def check_amendment_paragraph(self, model: Dict[str, Any], collection: str) -> None:
+        if collection != "motion":
+            return
+        if "amendment_paragraph" in model:
+            msg = f"{collection}/{model['id']}/amendment_paragraph error: "
+            if not isinstance(model["amendment_paragraph"], dict):
+                self.errors.append(msg + "Not a dict.")
+            for key, html in model["amendment_paragraph"].items():
+                if not str.isdigit(key):
+                    self.errors.append(msg + f"{key} is not a number.")
```
Comment (jsangmeister) [inline openslides_backend/models/checker.py:576]: also use the jsonschema here

File: openslides_backend/models/checker.py [hunk 4]
```diff
@@ -562,6 +564,21 @@ def get_enum_from_collection_field(
         field_type = self.get_model(collection).get_field(field)
         return field_type.constraints.get("enum")
 
+    def check_amendment_paragraph(self, model: Dict[str, Any], collection: str) -> None:
```
Comment (jsangmeister) [inline openslides_backend/models/checker.py:567]: Can you rename this to something like `check_special_fields` and move the recommendation extension validation code from https://github.com/OpenSlides/openslides-backend/blob/d30f2a82f49176442b8ebc1c0ce15df8c87b414f/openslides_backend/models/checker.py#L684-L705 also in here to unify the handling of such fields with extra validation?

**Review by jsangmeister — CHANGES_REQUESTED** [review]

File: openslides_backend/action/actions/motion/create.py (hunk 2 above)
Comment (reiterl) [reply openslides_backend/action/actions/motion/create.py:47]: Done.

File: openslides_backend/models/checker.py (hunk 3 above)
Comment (reiterl) [reply openslides_backend/models/checker.py:576]: Done.

File: openslides_backend/models/checker.py (hunk 4 above)
Comment (reiterl) [reply openslides_backend/models/checker.py:567]: Done.

**Review by jsangmeister — APPROVED** [review]

**Pull request merged**